client/src/pages/EditProfile/EditProfile.tsx

```diff
@@ -13,137 +12,225 @@ import {
   Paper,
   Box,
 } from '@material-ui/core';
-import DateFnsUtils from '@date-io/date-fns';
 import useStyles from './useStyles';
-import { DatePicker, MuiPickersUtilsProvider } from '@material-ui/pickers';
+import { FormikHelpers, useFormik } from 'formik';
+import * as yup from 'yup';
+import createPprofile, { IProfile } from '../../helpers/APICalls/createProfile';
+import { useSnackBar } from '../../context/useSnackbarContext';
+
+const validationSchema = yup.object({
+  firstName: yup.string().required('First Name is required'),
+  lastName: yup.string().required('Last Name is required'),
+  phone: yup.string(),
+  address: yup.string().required('Address is required'),
+  description: yup.string().required('Your description is required'),
+});
 
 export default function EditProfile(): JSX.Element {
   const classes = useStyles();
-  const [selectedDate, setSelectedDate] = useState(new Date());
-  const [gender, setGender] = useState('male');
-  const optionSelect = [];
-  for (let i = 1; i < 32; i++) {
-    optionSelect.push(
-      <MenuItem value={i} key={i}>
-        {i}
-      </MenuItem>,
-    );
-  }
+  const { updateSnackBarMessage } = useSnackBar();
+
+  const [fetchedProfile, setFetchedProfile] = useState({
+    firstName: '',
+    lastName: '',
+    gender: 'male',
+    birth: '',
+    email: '',
+    phone: '',
+    address: '',
+    description: '',
+  });
+
+  useEffect(() => {
+    fetch(`/profile`)
+      .then((response) => response.json())
+      .then((data) => setFetchedProfile(data));
+  }, []);
+
+  const handleSubmit = (
+    { firstName, lastName, email, phone, address, gender, birth, description }: IProfile,
+    { setSubmitting }: FormikHelpers<IProfile>,
+  ) => {
+    createPprofile(firstName, lastName, email, phone, address, gender, birth, description).then((data) => {
```

Comment:
On each form submission aren't we creating a new profile for the user? You can update the profile instead, if it already exists :)